airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -26,9 +26,84 @@
+from pipelines.models.secrets import Secret, SecretNotFoundError, SecretStore
 from pipelines.models.steps import STEP_PARAMS, MountPath, Step, StepResult, StepStatus
 
 
+def get_secrets_from_connector_test_suites_option(
+    connector_test_suites_options: List[Dict[str, str | Dict[str, List[Dict[str, str | Dict[str, str]]]]]],
+    suite_name: str,
+    secret_stores: Dict[str, SecretStore],
+    raise_on_missing_secret_store: bool = True,
+    logger: logging.Logger | None = None,
+) -> List[Secret]:
+    """Get secrets declared in metadata connectorTestSuitesOptions for a test suite name.
+    It will use the secret store alias declared in connectorTestSuitesOptions.
+    If the secret store is not available a warning or and error could be raised according to the raise_on_missing_secret_store parameter value.
+    We usually want to raise an error when running in CI context and log a warning when running locally, as locally we can fallback on local secrets.
+
+    Args:
+        connector_test_suites_options (List[Dict[str, str  |  Dict]]): The connector under test test suite options
+        suite_name (str): The test suite name
+        secret_stores (Dict[str, SecretStore]): The available secrets stores
+        raise_on_missing_secret_store (bool, optional): Raise an error if the secret store declared in the connectorTestSuitesOptions is not available. Defaults to True.
+        logger (logging.Logger | None, optional): Logger to log a warning if the secret store declared in the connectorTestSuitesOptions is not available. Defaults to None.
+
+    Raises:
+        SecretNotFoundError: Raised  if the secret store declared in the connectorTestSuitesOptions is not available and raise_on_missing_secret_store is truthy.
+
+    Returns:
+        List[Secret]: List of secrets declared in the connectorTestSuitesOptions for a test suite name.
+    """
+    secrets: List[Secret] = []
+
+    for enabled_test_suite in connector_test_suites_options:
+        if enabled_test_suite["suite"] == suite_name:
+            if enabled_test_suite.get("testSecrets"):
+                assert isinstance(enabled_test_suite["testSecrets"], list)
+                suite_secrets: List[Dict[str, str | Dict[str, str]]] = enabled_test_suite["testSecrets"]
+                for s in suite_secrets:
+                    if s["secretStore"]["alias"] not in secret_stores:
+                        message = f"Secret {s['name']} can't be retrieved as {s['secretStore']['alias']} is not available"
+                        if raise_on_missing_secret_store:
+                            raise SecretNotFoundError(message)
+                        if logger is not None:
+                            logger.warn(message)
+                        continue
+                    secret_store = secret_stores[s["secretStore"]["alias"]]
+                    secret = Secret(s["name"], secret_store, file_name=s.get("fileName"))
+                    secrets.append(secret)
+
+    return secrets
```

Comment:
💅 The nesting is too deep here.

I'd recommend we reduce it by
1. Using `find` instead of `for`
2. Using helper functions 
3. Inverting to remove the continue.


```suggestion
from pydash import find

def _handle_missing_secret_store(secret_info, raise_on_missing, logger):
    message = f"Secret {secret_info['name']} can't be retrieved as {secret_info['secretStore']['alias']} is not available"
    if raise_on_missing:
        raise SecretNotFoundError(message)
    if logger is not None:
        logger.warn(message)

def _process_secret(secret_info, secret_stores, raise_on_missing, logger):
    secret_store_alias = secret_info["secretStore"]["alias"]
    if secret_store_alias not in secret_stores:
        _handle_missing_secret_store(secret_info, raise_on_missing, logger)
        return None
    else:
        return Secret(secret_info["name"], secret_store, file_name=secret_info.get("fileName"))

def get_secrets_from_connector_test_suites_option(
    connector_test_suites_options: List[Dict[str, str | Dict[str, List[Dict[str, str | Dict[str, str]]]]]],
    suite_name: str,
    secret_stores: Dict[str, SecretStore],
    raise_on_missing_secret_store: bool = True,
    logger: logging.Logger | None = None,
) -> List[Secret]:
    """Get secrets declared in metadata connectorTestSuitesOptions for a test suite name.
    It will use the secret store alias declared in connectorTestSuitesOptions.
    If the secret store is not available a warning or and error could be raised according to the raise_on_missing_secret_store parameter value.
    We usually want to raise an error when running in CI context and log a warning when running locally, as locally we can fallback on local secrets.

    Args:
        connector_test_suites_options (List[Dict[str, str  |  Dict]]): The connector under test test suite options
        suite_name (str): The test suite name
        secret_stores (Dict[str, SecretStore]): The available secrets stores
        raise_on_missing_secret_store (bool, optional): Raise an error if the secret store declared in the connectorTestSuitesOptions is not available. Defaults to True.
        logger (logging.Logger | None, optional): Logger to log a warning if the secret store declared in the connectorTestSuitesOptions is not available. Defaults to None.

    Raises:
        SecretNotFoundError: Raised  if the secret store declared in the connectorTestSuitesOptions is not available and raise_on_missing_secret_store is truthy.

    Returns:
        List[Secret]: List of secrets declared in the connectorTestSuitesOptions for a test suite name.
    """
    secrets: List[Secret] = []
    enabled_test_suite = find(connector_test_suites_options, lambda x: x["suite"] == suite_name)

    if enabled_test_suite and "testSecrets" in enabled_test_suite:
        suite_secrets = enabled_test_suite["testSecrets"]
        for s in suite_secrets:
            valid_secret = _process_secret(s, secret_stores, raise_on_missing_secret_store, logger)
            if valid_secret:
                secrets.append(valid_secret)

    return secrets
```

Reply:
Tbh I actually think I prefer the original version 😄 . In particular, I'm wary of functions that return an `Optional[thing]` like `_process_secret` since it adds extra work to handle the `None` case.

This version also confirms my theory from my previous comment that getting rid of `raise_on_missing_secret_store` will make the code much nicer to read.